### PR TITLE
Week10 nbs upload trial 2

### PR DESCRIPTION
**Pull request by noahblakesmith**
Looking forward to your feedback. I am especially interested to hear if you have any thoughts on improving my algorithm for question 2. Thanks!

### CODE REVIEW

File: Noah - f2pHgty8hw/Week10_Assignment.do [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+	
+	cap drop longitude0 latitude0 distance longitude_* latitude_* group length height area criterion // Drop variables from previous loop
+	
+	*Choose random coordinate
+	local latitude0 = runiform(-90,90)
+	local longitude0 = runiform(-180,180)
+	gen latitude0 = `latitude0'
+	gen longitude0 = `longitude0'
+
+	geodist latitude longitude latitude0 longitude, gen(distance) // Calculate distance of each village to mutual point
```
Comment (AaronShtilerman) [inline Noah - f2pHgty8hw/Week10_Assignment.do]:
```suggestion
	geodist latitude longitude latitude0 longitude0, gen(distance) // Calculate distance of each village to mutual point
```

File: Noah - f2pHgty8hw/Week10_Assignment.do [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+local i = 0 // Counter for loop below
+
+while j[1] > 74100 { // Arbitrary threshold I chose that seemed reasonable
+	
+	local i = `i' + 1 // Update then display loop number
+	di `i'
+	
+	cap drop longitude0 latitude0 distance longitude_* latitude_* group length height area criterion // Drop variables from previous loop
+	
+	*Choose random coordinate
```
Comment (AaronShtilerman) [inline Noah - f2pHgty8hw/Week10_Assignment.do:81]: That's a pretty clever way to do things, I just picked points and took the 5 other points closest to them.

File: Noah - f2pHgty8hw/Week10_Assignment.do [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+decode b06_departemen, generate(department)
+merge m:1 department using `density'
+
+*Check for errors
+tabulate department if _merge == 1 // No observations in civ_section0 are unmatched_
+tabulate department if _merge == 2 // One district not in civ_section0 called "gbeleban" (problematic but not world-ending)
+
+*Outro
+compress
+save "question1.dta", replace
```
Comment (AaronShtilerman) [inline Noah - f2pHgty8hw/Week10_Assignment.do:58]: Looks great, good work for question 1

File: Noah - f2pHgty8hw/Week10_Assignment.do [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+append using `master'
+
+*Define and apply value label
+la def ward_category_label 1 "In 2010 only" 2 "In 2015 only" 3 "In 2010 and 2015"
+la val ward_category "ward_category_label"
+
+*Check for errors
+count if ward_category==1 | ward_category==3 // Equals 3,333 as expected
+count if ward_category==2 | ward_category==3 // Equals 3,944 as expected
```
Comment (AaronShtilerman) [inline Noah - f2pHgty8hw/Week10_Assignment.do:241]: This is a very thorough solution. I'm impressed!

**Review by AaronShtilerman — COMMENTED** [review]
Great work overall! Your sorting algorithm in 2 was very clever, aside from what I assume was a very small typo. Aside from that, I don't really have any other suggestions for how things could be improved!

File: Noah - f2pHgty8hw/Week10_Assignment.do (hunk 1 above)
Comment (noahblakesmith) [reply Noah - f2pHgty8hw/Week10_Assignment.do]: Nice catch. Thanks!

**Review by Aaron-Shtilerman — APPROVED** [review]
Great work!

**Pull request closed without merging**